pkg/template/event_test.go

```diff
@@ -920,6 +935,37 @@ func Test_NewResources(t *testing.T) {
 				json.RawMessage(`{"rt3": "default4"}`),
 			},
 		},
+		{
+			name: "one resource template multiple bindings",
+			args: args{
+				body:   json.RawMessage(`{"foo": "bar"}`),
+				header: map[string][]string{"one": {"1"}},
+				binding: ResolvedBinding{
+					TriggerTemplate: bldr.TriggerTemplate("tt", "namespace",
+						bldr.TriggerTemplateSpec(
+							bldr.TriggerTemplateParam("param1", "description", ""),
+							bldr.TriggerTemplateParam("param2", "description", ""),
+							bldr.TriggerResourceTemplate(json.RawMessage(`{"rt1": "$(params.param1)-$(params.param2)"}`)),
+						),
+					),
+					TriggerBinding: []*triggersv1.TriggerBinding{
+						bldr.TriggerBinding("tb", "namespace",
+							bldr.TriggerBindingSpec(
+								bldr.TriggerBindingParam("param1", "$(body.foo)"),
+							),
+						),
+						bldr.TriggerBinding("tb2", "namespace",
+							bldr.TriggerBindingSpec(
+								bldr.TriggerBindingParam("param2", "$(header.one)"),
+							),
+						),
+					},
+				},
+			},
+			want: []json.RawMessage{
+				json.RawMessage(`{"rt1": "bar-1"}`),
+			},
+		},
```

Comment:
maybe not something you want to tackle now @wlynch but i think unit testing at the level of `NewResources` is leading to the test cases being a long list of pretty involved setup + expected state; those kinda tests are hard to read and maintain over time, so i think we should start preferring more focused tests - e.g. test an example or end to end test that uses multiple bindings, then have unit tests for individual functions such as `mergeBindingParams`

Reply:
> * Could the docs talk about this feature, and why you'd want to use it?

Done.

> * The commit message (and probably the relase note too), I think they could include [more details](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages) around _why_ we want to allow multiple trigger bindings (e.g. use cases we want to support) (the phrase "multiple bindings" may not mean much to folks without more info)

Done

> And one question: do you know what behavior is if someone tried to use an event listener still using `binding` instead of `bindings`? I'm just thinking about our dogfooding, even tho we're still very very alpha, i wonder if it would make sense to raise an error in that case so we dont just get unexpected behavior we need to debug (maybe this is how it works already?)

There's validation before the resources are accepted, so you get an error like:
```
Error from server (BadRequest): error when creating "eventlistener.yaml": admission webhook "triggers-webhook.tekton.dev" denied the request: mutation failed: invalid value: triggerbindings.tekton.dev "pipeline-binding" not found: spec.triggers[0].binding.name
```